storage/core/src/test/java/com/hortonworks/registries/storage/transaction/ManagedTransactionTestHelper.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2017 Hortonworks.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package com.hortonworks.registries.storage.transaction;
+
+import com.hortonworks.registries.storage.exception.IgnoreTransactionRollbackException;
+
+public class ManagedTransactionTestHelper {
+    public static final Exception INTENDED_EXCEPTION = new Exception("Intended exception!");
+    public static final IgnoreTransactionRollbackException IGNORE_ROLLBACK_EXCEPTION =
+            new IgnoreTransactionRollbackException(INTENDED_EXCEPTION);
+
+    private boolean called = false;
+    private Object[] calledArgs;
+
+    public void call(Object...args) {
+        this.calledArgs = args;
+        called = true;
+    }
+
+    public void throwException(Object...args) throws Exception {
+        call(args);
+        throw INTENDED_EXCEPTION;
+    }
+
+    public void throwIgnoreRollbackException(Object...args) throws Exception {
+        call(args);
+        throw IGNORE_ROLLBACK_EXCEPTION;
```

Comment:
nit: better to throw desired Exception instead of using an instance which fills the stacktrace when this instance is created. I understand that stacktrace is not used in assertion but it may be  good to avoid using a created Exception.

Reply:
Addressed. I don't replace INTENDED_EXCEPTION since it is used as comparison. (I guess you are aware of this, but just be clear. :) )

Reply:
I am aware of that but I prefer to create a specific Exception class and check for that class instead of an `Exception` instance.

Reply:
Makes sense. Will address too.

Reply:
Address. Please take a look again at the latest commit. (Note: commit amended)